R/stao/CFEC_BIT_processing_199.R

```diff
@@ -0,0 +1,67 @@
+################
+#Issue #199: CFEC Basic Information Table (BIT)
+#Data Processing
+#February 2018
+#Sophia Tao
+################
+
+
+
+# read in
+bit <- read.csv("/home/stao/my-sasap/199_CFEC/BIT.csv", header = T)
+
+# delete unnecessary column
+bit$X....Preliminary <- NULL
+
+# replace "." with "NA"
+bit$Average.Permit.Price[bit$Average.Permit.Price=="."] <- NA
+bit$Total.Permits.Fished[bit$Total.Permits.Fished=="."] <- NA
+bit$Resident.Total.Pounds[bit$Resident.Total.Pounds=="."] <- NA
+bit$Nonresident.Total.Pounds[bit$Nonresident.Total.Pounds=="."] <- NA
+bit$Total.Pounds[bit$Total.Pounds=="."] <- NA
+bit$Resident.Average.Pounds[bit$Resident.Average.Pounds=="."] <- NA
+bit$Nonresident.Average.Pounds[bit$Nonresident.Average.Pounds=="."] <- NA
+bit$Average.Pounds[bit$Average.Pounds=="."] <- NA
+bit$Resident.Total.Earnings[bit$Resident.Total.Earnings=="."] <- NA
+bit$Nonresident.Total.Earnings[bit$Nonresident.Total.Earnings=="."] <- NA
+bit$Total.Earnings[bit$Total.Earnings=="."] <- NA
+bit$Resident.Average.Earnings[bit$Resident.Average.Earnings=="."] <- NA
+bit$Nonresident.Average.Earnings[bit$Nonresident.Average.Earnings=="."] <- NA
+bit$Average.Earnings[bit$Average.Earnings=="."] <- NA
+bit$Average.Permit.Price[bit$Average.Permit.Price=="."] <- NA
+
+# remove commas
+bit$Resident.Interim.Permits.Issued <- gsub(",", "", bit$Resident.Interim.Permits.Issued)
+bit$Resident.Interim.Permits.Issued <- gsub(",", "", bit$Resident.Interim.Permits.Issued)
```

Comment:
here's one example alternative method:

```
df <- data.frame(x = c("a", "1,x", "b", "c"),
        y = c("b,b", "a,", "b", "f"),
        z = c("a", "a", "g,,fhj", "a"),
        stringsAsFactors = FALSE)

sapply(df, function(col){gsub(",", "", col)})
```
This way returns your data frame as a matrix, so it's worth checking to make sure it doesn't mess anything else up. There are also tidyverse ways of doing things like this, but I'll leave that for you to explore :)

Reply:
this is really cool